Sources/MediaViewer/Transition/PresentationController.swift

```diff
@@ -1,34 +1,91 @@
 import UIKit
 
+struct PresentationConsts {
+    static let transitionViewTag = 602
+    static let backgroundViewTag = 601
+}
+
+@MainActor
 final class PresentationController: UIPresentationController {
+    
+    var onWillDismiss:(() -> Void)?
+    
     override func presentationTransitionWillBegin() {
-        containerView?.backgroundColor = .clear
-        containerView?.addSubview(presentedView!)
-        let previewController = presentedViewController as! PreviewController
-        
-        previewController
-            .topView?
-            .alpha = 0
-        
-        let fromTransition: CGAffineTransform
-        let fromScale: Double
-        let transitionView = previewController.currentTransitionView
-        if let containerView, let transitionView {
-            let point = transitionView
-                .convert(transitionView.bounds, to: containerView)
-                .applying(.init(
-                    translationX: -containerView.center.x,
-                    y: -containerView.center.y
-                ))
-            fromTransition = CGAffineTransform(translationX: point.midX, y: point.midY)
-            fromScale = transitionView.bounds.width / containerView.bounds.width
+        
+        guard let previewController = presentedViewController as? PreviewController,
+              let containerView,
+              let transitionView = previewController.currentTransitionView,
+              let topView = previewController.topView else { return }
+        
+        containerView.backgroundColor = .clear
+        containerView.addSubview(presentedView!)
```

Comment:
Can we find a way to make sure presentedView exist? I don't think this is worth crashing for, we could probably handle it.